xmldumps-backup/mwbzutils/docs/checkforbz2footer.1
.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.40.4.
.TH CHECKFORBZ2FOOTER "1" "July 2013" "checkforbz2footer 0.0.4" "User Commands"
.SH NAME
checkforbz2footer \- Check if bzip2 file ends with bz2 magic footer
.SH SYNOPSIS
.B checkforbz2footer
[\fI--version|--help\fR]
.br
.B checkforbz2footer
\fI<infile>\fR
.SH DESCRIPTION
Check whether the specified bzip2 compressed file ends with a bz2 footer
or not ((i.e. if it is truncated or corrupted).
This is a crude but fast test for integrity; we don't check the CRC at
the end of the stream, nor do we check the bit padding in the last byte
of the file.
.PP
Exits with 0 if the file has the bz2 footer, 1 if the file does not have
the footer and \fB\-1\fR on error.
.SH OPTIONS

Flags:
.TP
\fB\-h\fR, \fB\-\-help\fR
Show this help message
.TP
\fB\-v\fR, \fB\-\-version\fR
Display the version of this program and exit
.PP
Arguments:
.TP
<infile>
Name of the file to check
.SH AUTHOR
Written by Ariel T. Glenn.
.SH "REPORTING BUGS"
Report bugs in checkforbz2footer to <https://bugzilla.wikimedia.org/>.
.PP
.br
See also:
.IP
dumpbz2filefromoffset(1), dumplastbz2block(1), findpageidinbz2xml(1)
recompressxml(1), writeuptopageid(1)
.SH COPYRIGHT
Copyright \(co 2011, 2012, 2013 Ariel T. Glenn.  All rights reserved.
.PP
This program is free software: you can redistribute it and/or modify it
under the  terms of the GNU General Public License as published by the
Free Software Foundation, either version 2 of the License, or (at your
option) any later version.
.PP
This  program  is  distributed  in the hope that it will be useful, but
WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
Public License for more details.
.PP
You should have received a copy of the GNU General Public License along
with this program.  If not, see <http://www.gnu.org/licenses/>


xmlfileutils/CHANGELOG
29 May 2013 Kent L. Miller <wpmirrordev@gmail.com>

   Release 0.0.2-1

    * mwxml2sql.c:
      Doc: show_version() rewritten for compatibility with help2man.
      Doc: usage() rewritten for compatibility with help2man.
      Doc: man page generated with help2man
    * sql2txt.c:
      Bug fixed: long options, --sqlfile and --textfile, did not work.
      Doc: show_version() written for compatibility with help2man.
      Doc: usage() rewritten for compatibility with help2man.
      Doc: man page generated with help2man
      Option: added -V, --version option for use with help2man
    * sqlfilter.c:
      Doc: show_version() written for compatibility with help2man.
      Doc: usage() rewritten for compatibility with help2man.
      Doc: man page generated with help2man
      Option: created -w, --version option for use by help2man
    * Makefile:
      Debian: build and deinstall targets written; clean target improved
      Debian: install target rewritten to use: install $(DESTDIR)/...
      Debian: dist renamed `mediawiki-mwxml2sql-0.0.2.tar.gz'
      Doc: man pages generated using `help2man'
      MWXML2SQLVERS for Makefile dist version and for VERSION in source
    * mwxml2sql.h:
      VERSION passed from Makefile

22 May 2013 Ariel T. Glenn <ariel@wikimedia.org>

    * mwxmlelts.c:
      handle text with <deleted /> instead of content
      workaround for XML files that may contain duplicate text ids

21 May 2013 Ariel T. Glenn <ariel@wikimedia.org>

    * mwxmlelts.c:
      convert revision timestamp from datestring format to db format

10 Apr 2013 Ariel T. Glenn <ariel@wikimedia.org>

    Release 0.0.2

    * sqlfilter: filters sql table dumps against e.g. list of pageids
      (and the column number of the pageid in the sql row) and writes
      only those tuples that match
    * wikiretriever.py: retrieves page titles and or content from a wiki
    * wikicontent2sql.py: downloads all pages containing a given template,
      swapping out talk pages for their corresponding pages (eg File talk
      for File etc.), converts them to page, revision and table sql,
      and filters downloaded sql table dumps according to the page ids
      of the downloaded content, producing a pile of sql files for import
    * package: various bugfixes

25 Jan 2013 Ariel T. Glenn <ariel@wikimedia.org>

   Release 0.0.1

   Features:
    *  Reads stub and text content xml files in parallel, producing separate
       sql files for inserts to page, revision and text tables
    *  Reads bz2, gz or plain text content, writes bz2, gz or plaintext output
       based on extensions of stub/page content files and output filename
    *  Reads xml dumps with schema 0.5 through 0.8, writes for MW 1.5 through
       1.20 (tested only for 0.7 in and MW 1.20 out)


xmldumps-backup/mwbzutils/checkforbz2footer.c
#include <unistd.h>
#include <stdio.h>
#include <string.h>
#include <sys/types.h>
#include <sys/stat.h>
#include <fcntl.h>
#include <stdlib.h>
#include <errno.h>
#include <getopt.h>
#include "mwbzutils.h"

void usage(char *message) {
  char * help =
"Usage: checkforbz2footer [--version|--help]\n"
"   or: checkforbz2footer <infile>\n\n"
"Check whether the specified bzip2 compressed file ends with a bz2 footer\n"
"or not ((i.e. if it is truncated or corrupted).\n"
"This is a crude but fast test for integrity; we don't check the CRC at\n"
"the end of the stream, nor do we check the bit padding in the last byte\n"
"of the file.\n\n"
"Exits with 0 if the file has the bz2 footer, 1 if the file does not have\n"
"the footer and -1 on error.\n\n"
"Options:\n\n"
"Flags:\n\n"
"  -h, --help       Show this help message\n"
"  -v, --version    Display the version of this program and exit\n\n"
"Arguments:\n\n"
"  <infile>         Name of the file to check\n\n"
"Report bugs in checkforbz2footer to <https://bugzilla.wikimedia.org/>.\n\n"
"See also:\n\n"
"  dumpbz2filefromoffset(1), dumplastbz2block(1), findpageidinbz2xml(1)\n"
"  recompressxml(1), writeuptopageid(1)\n\n";
  if (message) {
    fprintf(stderr,"%s\n\n",message);
  }
  fprintf(stderr,"%s",help);
  exit(-1);
}

void show_version(char *version_string) {
  char * copyright =
"Copyright (C) 2011, 2012, 2013 Ariel T. Glenn.  All rights reserved.\n\n"
"This program is free software: you can redistribute it and/or modify it\n"
"under the  terms of the GNU General Public License as published by the\n"
"Free Software Foundation, either version 2 of the License, or (at your\n"
"option) any later version.\n\n"
"This  program  is  distributed  in the hope that it will be useful, but\n"
"WITHOUT ANY WARRANTY; without even the implied warranty of \n"
"MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General\n"
"Public License for more details.\n\n"
"You should have received a copy of the GNU General Public License along\n"
"with this program.  If not, see <http://www.gnu.org/licenses/>\n\n"
    "Written by Ariel T. Glenn.\n";
  fprintf(stderr,"checkforbz2footer %s\n", version_string);
  fprintf(stderr,"%s",copyright);
  exit(-1);
}

int main(int argc, char **argv) {

  int fin;
  int result;
  bz_info_t bfile;

  int optc;
  int optindex=0;

  struct option optvalues[] = {
    {"help", 0, 0, 'h'},
    {"version", 0, 0, 'v'},
    {NULL, 0, NULL, 0}
  };

  if (argc != 2) {
    usage("Missing option or argument.");
    exit(-1);
  }

  while (1) {
    optc=getopt_long_only(argc,argv,"hv", optvalues, &optindex);
    if (optc=='h')
      usage(NULL);
    else if (optc=='v')
      show_version(VERSION);
    else if (optc==-1) break;
    else usage("Unknown option or other error\n");
  }

  if (optind >= argc) {
    usage("Missing filename argument.");
  }

  fin = open (argv[optind], O_RDONLY);
  if (fin < 0) {
    fprintf(stderr,"failed to open file %s for read\n", argv[optind]);
    exit(-1);
  }

  bfile.footer = init_footer();
  result = check_file_for_footer(fin, &bfile);
  close(fin);
  if (result == -1) {
    return(result);
  }
  else {
    return(0);
  }
}

